tests/run-make/prefer-dylib/rmake.rs

```diff
@@ -0,0 +1,20 @@
+//@ ignore-cross-compile
+
+use run_make_support::{cwd, read_dir, run, run_fail, rustc};
+use std::fs::remove_file;
+use std::process::Command;
+
+fn main() {
+    rustc().input("bar.rs").crate_type("dylib").crate_type("rlib").arg("-Cprefer-dynamic").run();
+    rustc().input("foo.rs").arg("-Cprefer-dynamic").run();
+
+    run("foo");
+
+    read_dir(cwd(), |path| {
```

Comment:
I know that this was in the original Makefile, but I think that we can improve it to this:
```rust
remove_file(dynamic_lib_name("bar")).unwrap();
```
To make it easier to understand what's going on. Removing the dynamic library is enough to keep the test working.